src/compass/rdr2geo.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+
+'''wrapper for rdr2geo'''
+
+import os
+import time
+
+import isce3
+import journal
+from osgeo import gdal
+
+from compass.utils.runconfig import RunConfig
+from compass.utils.yaml_argparse import YamlArgparse
+
+
+def run(cfg):
+    '''run rdr2geo with provided runconfig'''
+    info_channel = journal.info("rdr2geo.run")
+
+    t_start = time.time()
+    info_channel.log("starting rdr2geo")
+
+    # common rdr2geo inits
+    dem_raster = isce3.io.Raster(cfg.dem)
+    epsg = dem_raster.get_epsg()
+    proj = isce3.core.make_projection(epsg)
+    ellipsoid = proj.ellipsoid
+
+    # check if gpu ok to use
+    use_gpu = isce3.core.gpu_check.use_gpu(cfg.gpu_enabled, cfg.gpu_id)
+    if use_gpu:
+        # Set the current CUDA device.
+        device = isce3.cuda.core.Device(cfg.gpu_id)
+        isce3.cuda.core.set_device(device)
+
+    # save SLC for all bursts
+    # run rdr2geo for only once per burst_id
+    for bursts in cfg.bursts:
+        # init output directory in product_path
+        output_path = f'{cfg.product_path}/{bursts[0].burst_id}'
+        os.makedirs(output_path, exist_ok=True)
+
+        # save SLC to ENVI for all bursts
+        for burst in bursts:
+            burst.slc_to_file(f'{output_path}/{burst.polarization}.slc')
+
+        # run rdr2geo for only 1 burst avoid redundancy
+        # get isce3 objs from burst
+        rdr_grid = burst.as_isce3_radargrid()
+        isce3_orbit = burst.orbit
+
+        # init grid doppler
+        grid_doppler = isce3.core.LUT2d()
+
+        # init CPU or CUDA object accordingly
+        if use_gpu:
+            Rdr2Geo = isce3.cuda.geometry.Rdr2Geo
+        else:
+            Rdr2Geo = isce3.geometry.Rdr2Geo
+
+        # init rdr2geo obj
+        rdr2geo_obj = Rdr2Geo(
+            rdr_grid,
+            isce3_orbit,
+            ellipsoid,
+            grid_doppler)
+
+        # set rdr2geo params
+        for key, val in cfg.rdr2geo_params.__dict__.items():
+            setattr(rdr2geo_obj, key, val)
+
+        # prepare output rasters
+        topo_output = {'x':(True, gdal.GDT_Float64),
+                       'y':(True, gdal.GDT_Float64),
+                       'z':(True, gdal.GDT_Float64),
+                       'layoverShadowMask':(cfg.rdr2geo_params.compute_mask,
+                                            gdal.GDT_Byte)}
+        raster_list = [
+            isce3.io.Raster(f'{output_path}/{fname}.rdr', rdr_grid.width,
+                            rdr_grid.length, 1, dtype, 'ENVI')
+            if enabled else None
+            for fname, (enabled, dtype) in topo_output.items()]
+        x_raster, y_raster, z_raster, layover_shadow_raster = raster_list
+
+        # run rdr2geo
+        rdr2geo_obj.topo(dem_raster, x_raster, y_raster, z_raster,
```

Comment:
Do not forget to set the EPSG of the topo.vrt. This will be necessary for the sub-sequent module to properly work

Reply:
Output EPSG defaults to `4326` thru [pybind rdr2geo](https://github.com/isce-framework/isce3/blob/73673f757cc30d0d399e9233ecec9008a6be319f/python/extensions/pybind_isce3/geometry/rdr2geo.cpp#L179).

I'm going to add optional `epsg` schema field in `product_path_group`. The default yaml value will be set to 4326.

Reply:
That was not my experience. The ESPG of the generated output VRT was left unset (i.e. -9999). The following lines:
```
output_vrt = isce3.io.Raster(f'{output_path}/topo.vrt', raster_list)
output_vrt.set_epsg(rdr2geo_obj.epsg_out)
```
solved my issue. I do not think is necessary to expose the EPSG in the yaml.

Reply:
`rdr2geo_obj` defaults to `epsg` of 4236. There's no way to change that in the code right now. Is that ok?

Reply:
Yeah, I am fine with `4326`. My only concern was that the EPSG is not carried over to the VRT (and this will cause `geo2rdr `to Seg fault)